src/core_plugins/apm_core/index.js

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+export default function apmCore(kibana) {
+  return new kibana.Plugin({
+    id: 'apm_core',
+
+    config(Joi) {
+      return Joi.object({
+        // enable plugin
+        enabled: Joi.boolean().default(true),
+
+        // Kibana Index pattern
+        indexPattern: Joi.string().default('apm-*'),
+
+        // ES Indices
+        errorIndices: Joi.string().default('apm-*-error-*'),
+        onboardingIndices: Joi.string().default('apm-*'), // TODO: change to "apm-*-onboarding-*" when APM-Server is ready
+        spanIndices: Joi.string().default('apm-*-span-*'),
+        transactionIndices: Joi.string().default('apm-*-transaction-*'),
+      }).default();
+    },
+  });
+}
```

Comment:
@epixa Created a new `apm_core` plugin with config vars. Looks ok?

Reply:
Seems fine to me at a glance!